packages/venia-ui/lib/components/OrderHistoryPage/orderHistoryPage.gql.js

```diff
@@ -0,0 +1,170 @@
+import { gql } from '@apollo/client';
+
+/*
+    This feature is being built ahead of GraphQL coverage that is landing in 2.4.1 of Magento. We're going to mock
+    the data based on the approved schema to make removing the mocking layer as seamless as possible.
+
+    @see https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/customer-orders.md
+ */
+
+/* eslint-disable graphql/template-strings */
+export const GET_CUSTOMER_ORDERS = gql`
+    query GetCustomerOrders {
+        customer {
+            id
+            orders {
+                items {
+                    id
+                    invoices {
+                        id
+                    }
+                    items {
+                        id
+                        product_sku
+                    }
+                    number
+                    order_date
+                    shipments {
+                        id
+                    }
+                    status
+                    total {
+                        grand_total {
+                            currency
+                            value
+                        }
+                    }
+                }
+            }
+        }
+    }
+`;
+
+export const OrderHistoryTypeDefs = gql`
+    extend type Customer {
+        orders: CustomerOrders
+    }
+
+    type CustomerOrders {
+        items: [CustomerOrder]!
+    }
+
+    type CustomerOrder {
+        id: ID!
+        order_date: String!
+        status: String!
+        number: String!
+        items: [OrderItemInterface]
+        total: OrderTotal
+        invoices: [Invoice]
+        shipments: [OrderShipment]
+    }
+
+    interface OrderItemInterface {
+        id: ID!
+        product_sku: String!
+    }
+
+    type OrderItem implements OrderItemInterface {
+        id: ID!
+        product_sku: String!
+    }
+
+    type OrderTotal {
+        grand_total: Money!
+    }
+
+    type Invoice {
+        id: ID!
+    }
+
+    type OrderShipment {
+        id: ID!
+    }
+`;
+/* eslint-enable graphql/template-strings */
+
+export const OrderHistoryResolvers = {
```

Comment:
I know we merged this in Wishlists but now that the v3 upgrade has been merged have you tried to do this with type policies and `useTypePolicies` hook?

Reply:
I have not, but was a partial reason for the draft PR 🙌  I merged mainline and only had to fix a couple imports, was going to ask you this morning if there's anything else I could fix to be more inline with v3. I'll give this a shot (and fix Wishlist if it goes smoothly).

Reply:
If you're only adding `@client` fields then you're probably OK but general advice to anyone working in the repo is to be on the lookout for apollo warnings such as normalization issues. v3 does a much better job of alerting us so hopefully we can catch issues quicker.

Additionally, when querying local fields I would say the main thing to be aware of is that there is a difference between an entirely client side query and a query that has both client and server props. 

```gql
export const GET_CUSTOMER_WISHLIST = gql`
    query GetCustomerWishlist {
        customer {
            id
            wishlists @client {
                id
                items_count
                name
                sharing_code
            }
        }
    }
`;
```

vs

```gql
export const GET_CUSTOMER_WISHLIST = gql`
    query GetCustomerWishlist {
        customer @client {
            id
            wishlists {
                id
                items_count
                name
                sharing_code
            }
        }
    }
`;
```

The former will always make a network request for `id` whereas the latter will not. [As written](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/WishlistPage/wishlistPage.gql.js#L15) you'll see a network request go out when you navigate to the customer wishlist page even though we don't need `id`. Of course, we may eventually need more fields like `name`, etc, so it might not matter.

Reply:
I experimented with this change and it introduced some odd behavior where data was only returned on the first request, every request after that was empty. Will need some help with this if this is the preferred pattern from Apollo.

Besides the above issue, I somewhat prefer the mixed usage and throwing `@client` on the field. This allows us to more easily test loading states while it's fetching `id`, giving us a more realistic render.